Source/GSCSConstraint.m

```diff
@@ -1,27 +1,366 @@
+              expression: expression
+                variable: nil];
+}
+
+- (instancetype) initEditConstraintWithVariable: (GSCSVariable *)variable
+                                        stength: (GSCSStrength *)strength
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: variable
+                                         coefficient: -1
+                                            constant: [variable value]];
+  GSCSConstraint *constraint = [self initWithType: GSCSConstraintTypeEdit
+                   strength: strength
+                 expression: expression
+                   variable: variable];
+  RELEASE(expression);
+  return constraint;
+}
+
+- (instancetype) initStayConstraintWithVariable: (GSCSVariable *)variable
+                                       strength: (GSCSStrength *)strength
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: variable
+                                         coefficient: -1
+                                            constant: [variable value]];
+  GSCSConstraint *constraint = [[GSCSConstraint alloc] initWithType: GSCSConstraintTypeStay
+                                     strength: strength
+                                   expression: expression
+                                     variable: variable];
+  RELEASE(expression);
+  return constraint;
+}
+
+- (instancetype) initWithLhsVariable: (GSCSVariable *)lhs
+                   equalsRhsVariable: (GSCSVariable *)rhs
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: lhs];
+  [expression addVariable: rhs coefficient: -1];
+  GSCSConstraint *constraint = [self initLinearConstraintWithExpression: expression];
+  RELEASE(expression);
+
+  return constraint;
+}
+
+- (instancetype) initWithLhsVariable: (GSCSVariable *)lhs
+                      equalsConstant: (CGFloat)rhs
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: lhs
+                                         coefficient: 1
+                                            constant: -rhs];
+  GSCSConstraint *constraint = [self initLinearConstraintWithExpression: expression];
+  RELEASE(expression);
+  return constraint;
+}
+
++ (GSCSConstraint *) constraintWithLeftVariable: (GSCSVariable *)lhs
+operator: (GSCSConstraintOperator) operator
+                                  rightVariable: (GSCSVariable *)rhsVariable
+{
+  if (operator== GSCSConstraintOperatorEqual)
+    {
+      GSCSLinearExpression *expression =
+          [[GSCSLinearExpression alloc] initWithVariable: lhs];
+      [expression addVariable: rhsVariable coefficient: -1];
+
+      RELEASE(expression);
+
+      return AUTORELEASE([[GSCSConstraint alloc]
+          initLinearConstraintWithExpression: expression]);
+    }
+  else
+    {
+      GSCSLinearExpression *rhsExpression =
+          [[GSCSLinearExpression alloc] initWithVariable: rhsVariable];
```

Comment:
Here you switched the role of rhs and lhs. This may still be correct, but it is a bit confusing. Maybe it would be better to implement all cases similar?